Makefile

```diff
@@ -40,12 +43,17 @@ ifdef RESULTS_DIR
 	RESULTS_VOLUME := -v $(RESULTS_DIR):/results$(VOLUME_FLAGS)
 endif
 
+ifneq ($(USE_GOCO),)
+	GO_CMD:=$(CONTAINER_CMD) run $(CONTAINER_OPTS) --rm $(GOCACHE_VOLUME) -v $(CURDIR):/go/src/github.com/ceph/go-ceph$(VOLUME_FLAGS) --entrypoint $(GO_CMD) $(CI_IMAGE_TAG)
+	GOFMT_CMD:=$(CONTAINER_CMD) run $(CONTAINER_OPTS) --rm $(GOCACHE_VOLUME) -v $(CURDIR):/go/src/github.com/ceph/go-ceph$(VOLUME_FLAGS) --entrypoint $(GOFMT_CMD) $(CI_IMAGE_TAG)
+endif
+
 build:
-	go build -v -tags $(BUILD_TAGS) $(shell go list ./... | grep -v /contrib)
+	$(GO_CMD) build -v -tags $(BUILD_TAGS) $(shell $(GO_CMD) list ./... | grep -v /contrib)
```

Comment:
This seems a bit heavy to start a container just to create the list of packages. Can't we use bash as entrypoint and then somehow get shell expression evaluation _in_ the container?

Reply:
Perhaps, but I was trying to start simple. Also, I have been doing this in practice for a few weeks and it seems to perform OK.  I'd be open to making it better in the future but would prefer to unblock the workflow first before optimizing it.

Reply:
And you don't want to hardcode just "go" there, because you want the same go version? Or is the makefile already independent of a local go install with this change?

Reply:
If I understand you correctly, I think it's the latter. That change is all about making the the "go" we use in the makefile independent of whatever "go" means to the system, allowing you to use a different but local go or, as I do in the followup patch, one within the container.